ckanext/datastore/backend/postgres.py

```diff
@@ -697,17 +728,20 @@ def _drop_indexes(context: Context, data_dict: dict[str, Any],
             AND t.relkind = 'r'
             AND idx.indisunique = {unique}
             AND idx.indisprimary = false
-            AND t.relname = %s
+            AND t.relname = :relname
         """.format(unique='true' if unique else 'false')
     indexes_to_drop = context['connection'].execute(
-        sql_get_index_string, data_dict['resource_id']).fetchall()
+        sa.text(sql_get_index_string),
+        {"relname": data_dict['resource_id']}
+    ).fetchall()
     for index in indexes_to_drop:
-        context['connection'].execute(
-            sql_drop_index.format(index[0]).replace('%', '%%'))
+        context['connection'].execute(sa.text(
+            sql_drop_index.format(index[0])
```

Comment:
potentially unsafe index name formatting here, but only if a bad name was created in the datastore db by another application/db user

Reply:
Fixed - `sa.column` should work here as well